pkg/virt-controller/watch/vm_test.go

```diff
@@ -1125,29 +1125,56 @@ var _ = Describe("VirtualMachine", func() {
 					Name: "dv1",
 				},
 			})
-
-			existingDataVolume, _ := watchutil.CreateDataVolumeManifest(virtClient, vm.Spec.DataVolumeTemplates[0], vm)
-
-			existingDataVolume.Namespace = "default"
-			existingDataVolume.Status.Phase = cdiv1.Succeeded
+			vm.Spec.Running = nil
+			vm.Spec.RunStrategy = &runStrategy
 
 			vm, err := virtFakeClient.KubevirtV1().VirtualMachines(vm.Namespace).Create(context.TODO(), vm, metav1.CreateOptions{})
 			Expect(err).To(Succeed())
+
+			shouldExpectDataVolumeCreation(vm.UID, map[string]string{"kubevirt.io/created-by": string(vm.UID)}, nil, pointer.P(0))
```

Comment:
@acardace This doesn't seem to guarantee the volume was created.

Reply:
The expect for the event does though.